### ISSUE TITLE
[Bug] gps is turned on and off too fast

### ISSUE THREAD

**Issue by Kright** [redfish64/TinyTravelTracker#93]
```
gps desire manager has asked for more than allowed time, gpsTimeAvailable: -15234, desireManager.currTimeWanted: 30000
```
Sometimes gpsTimeAvailable becomes negative, after that app swithing gps state between on/off very fast, which leads to a lot of logs and (may be) increased battery usage. 
In settings desired gs usage is 100%.

Wake log log on my sdcard was 2.5 Gb after a day, this isn't ok. Phone model: Samsung S8.

I have made a pull request: https://github.com/redfish64/TinyTravelTracker/pull/94

Logs:
```
Mode  9: gpsAttemptStartedFromPhoneBootMs -1561785997005 1970-01-13 06:54:19.402                 
2019-07-11 12:46:58.627 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295233 Mode  9: gpsAttemptEndedFromPhoneBootMs -1561785996990 1970-01-13 06:54:19.417                 
2019-07-11 12:46:58.628 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295234 Mode  9: lastGpsReadingFromBootMs       -1561786006753 1970-01-13 06:54:09.654                 
2019-07-11 12:46:58.629 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295235 Mode  9: lastGpsStatsUpdateFromPhoneBootMs -1561785996990 1970-01-13 06:54:19.417                 
2019-07-11 12:46:58.630 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295236 Mode  9: nextSignificantEventTime                         -1
2019-07-11 12:46:58.630 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859419 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.631 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295237 Mode  9: freeGpsTimeMs                                -15246
2019-07-11 12:46:58.632 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295238 Mode  9: totalTimeGpsRunningMs                       1743091
2019-07-11 12:46:58.633 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295239 Mode  9: totalTimeNotRunningGpsMs                     -15246
2019-07-11 12:46:58.633 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295240 Mode  9: totalSuccessfulGpsTries                          61
2019-07-11 12:46:58.634 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295241 Mode  9: totalGpsTries                                 11345
2019-07-11 12:46:58.635 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295242 Mode  9: longTimeWanted                                73639
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295243 Mode  9: shortTimeWanted                               30000
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps use perc is 1.0
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: Looping while not shutdown
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to enter synchronization block
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update strategy
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: deltaTimeMs 0
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859431 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050889431 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps desire manager has asked for more than allowed time, gpsTimeAvailable: -15234, desireManager.currTimeWanted: 30000
2019-07-11 12:46:58.636 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update gpsReader to true
2019-07-11 12:46:58.641 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to set gpsOn and writeUpdateStatus
2019-07-11 12:46:58.641 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil: Thread Thread[Thread-97,5,main] Mode MODE_WRITE_STRATEGY_STATUS
2019-07-11 12:46:58.642 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295244 Mode  9: gpsOn                          t
2019-07-11 12:46:58.642 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295245 Mode  9: gpsAttemptStartedFromPhoneBootMs -1561785996976 1970-01-13 06:54:19.431                 
2019-07-11 12:46:58.643 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295246 Mode  9: gpsAttemptEndedFromPhoneBootMs -1561785996990 1970-01-13 06:54:19.417                 
2019-07-11 12:46:58.644 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295247 Mode  9: lastGpsReadingFromBootMs       -1561786006753 1970-01-13 06:54:09.654                 
2019-07-11 12:46:58.645 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295248 Mode  9: lastGpsStatsUpdateFromPhoneBootMs -1561785996976 1970-01-13 06:54:19.431                 
2019-07-11 12:46:58.646 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295249 Mode  9: nextSignificantEventTime                     -15238
2019-07-11 12:46:58.646 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859435 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.647 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295250 Mode  9: freeGpsTimeMs                                -15230
2019-07-11 12:46:58.648 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295251 Mode  9: totalTimeGpsRunningMs                       1743091
2019-07-11 12:46:58.648 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295252 Mode  9: totalTimeNotRunningGpsMs                     -15230
2019-07-11 12:46:58.649 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295253 Mode  9: totalSuccessfulGpsTries                          61
2019-07-11 12:46:58.650 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295254 Mode  9: totalGpsTries                                 11346
2019-07-11 12:46:58.651 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295255 Mode  9: longTimeWanted                                73639
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295256 Mode  9: shortTimeWanted                               30000
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps use perc is 1.0
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: Looping while not shutdown
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to enter synchronization block
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update strategy
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: deltaTimeMs 0
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859446 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.652 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update gpsReader to false
2019-07-11 12:46:58.658 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to set gpsOn and writeUpdateStatus
2019-07-11 12:46:58.658 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil: Thread Thread[Thread-97,5,main] Mode MODE_WRITE_STRATEGY_STATUS
2019-07-11 12:46:58.659 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295257 Mode  9: gpsOn                          f
2019-07-11 12:46:58.660 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295258 Mode  9: gpsAttemptStartedFromPhoneBootMs -1561785996976 1970-01-13 06:54:19.431                 
2019-07-11 12:46:58.661 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295259 Mode  9: gpsAttemptEndedFromPhoneBootMs -1561785996961 1970-01-13 06:54:19.446                 
2019-07-11 12:46:58.662 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295260 Mode  9: lastGpsReadingFromBootMs       -1561786006753 1970-01-13 06:54:09.654                 
2019-07-11 12:46:58.663 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295261 Mode  9: lastGpsStatsUpdateFromPhoneBootMs -1561785996961 1970-01-13 06:54:19.446                 
2019-07-11 12:46:58.664 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295262 Mode  9: nextSignificantEventTime                         -6
2019-07-11 12:46:58.664 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859453 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.665 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295263 Mode  9: freeGpsTimeMs                                -15212
2019-07-11 12:46:58.666 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295264 Mode  9: totalTimeGpsRunningMs                       1743091
2019-07-11 12:46:58.666 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295265 Mode  9: totalTimeNotRunningGpsMs                     -15212
2019-07-11 12:46:58.667 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295266 Mode  9: totalSuccessfulGpsTries                          61
2019-07-11 12:46:58.668 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295267 Mode  9: totalGpsTries                                 11346
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295268 Mode  9: longTimeWanted                                73639
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295269 Mode  9: shortTimeWanted                               30000
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps use perc is 1.0
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 D/GpsTrailer: Looping while not shutdown
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to enter synchronization block
2019-07-11 12:46:58.669 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update strategy
2019-07-11 12:46:58.670 19624-20311/com.rareventure.gps2 D/GpsTrailer: deltaTimeMs 0
2019-07-11 12:46:58.670 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859464 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.670 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050889464 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.670 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps desire manager has asked for more than allowed time, gpsTimeAvailable: -15201, desireManager.currTimeWanted: 30000
2019-07-11 12:46:58.670 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update gpsReader to true
2019-07-11 12:46:58.684 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to set gpsOn and writeUpdateStatus
2019-07-11 12:46:58.684 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil: Thread Thread[Thread-97,5,main] Mode MODE_WRITE_STRATEGY_STATUS
2019-07-11 12:46:58.685 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295270 Mode  9: gpsOn                          t
2019-07-11 12:46:58.686 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295271 Mode  9: gpsAttemptStartedFromPhoneBootMs -1561785996943 1970-01-13 06:54:19.464                 
2019-07-11 12:46:58.687 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295272 Mode  9: gpsAttemptEndedFromPhoneBootMs -1561785996961 1970-01-13 06:54:19.446                 
2019-07-11 12:46:58.688 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295273 Mode  9: lastGpsReadingFromBootMs       -1561786006753 1970-01-13 06:54:09.654                 
2019-07-11 12:46:58.689 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295274 Mode  9: lastGpsStatsUpdateFromPhoneBootMs -1561785996943 1970-01-13 06:54:19.464                 
2019-07-11 12:46:58.690 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295275 Mode  9: nextSignificantEventTime                     -15216
2019-07-11 12:46:58.690 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859479 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.691 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295276 Mode  9: freeGpsTimeMs                                -15186
2019-07-11 12:46:58.691 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295277 Mode  9: totalTimeGpsRunningMs                       1743091
2019-07-11 12:46:58.692 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295278 Mode  9: totalTimeNotRunningGpsMs                     -15186
2019-07-11 12:46:58.693 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295279 Mode  9: totalSuccessfulGpsTries                          61
2019-07-11 12:46:58.694 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295280 Mode  9: totalGpsTries                                 11347
2019-07-11 12:46:58.695 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295281 Mode  9: longTimeWanted                                73639
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295282 Mode  9: shortTimeWanted                               30000
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: gps use perc is 1.0
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: Looping while not shutdown
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to enter synchronization block
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update strategy
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: deltaTimeMs 0
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859490 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.696 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to update gpsReader to false
2019-07-11 12:46:58.704 19624-20311/com.rareventure.gps2 D/GpsTrailer: About to set gpsOn and writeUpdateStatus
2019-07-11 12:46:58.704 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil: Thread Thread[Thread-97,5,main] Mode MODE_WRITE_STRATEGY_STATUS
2019-07-11 12:46:58.705 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295283 Mode  9: gpsOn                          f
2019-07-11 12:46:58.706 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295284 Mode  9: gpsAttemptStartedFromPhoneBootMs -1561785996943 1970-01-13 06:54:19.464                 
2019-07-11 12:46:58.707 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295285 Mode  9: gpsAttemptEndedFromPhoneBootMs -1561785996917 1970-01-13 06:54:19.490                 
2019-07-11 12:46:58.708 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295286 Mode  9: lastGpsReadingFromBootMs       -1561786006753 1970-01-13 06:54:09.654                 
2019-07-11 12:46:58.709 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295287 Mode  9: lastGpsStatsUpdateFromPhoneBootMs -1561785996917 1970-01-13 06:54:19.490                 
2019-07-11 12:46:58.710 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295288 Mode  9: nextSignificantEventTime                         -8
2019-07-11 12:46:58.710 19624-20311/com.rareventure.gps2 D/GpsTrailer: currTimeMs 1050859499 startTimeFromPhoneBoot 1049131574totalTimeGpsRunning 1743091
2019-07-11 12:46:58.711 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295289 Mode  9: freeGpsTimeMs                                -15166
2019-07-11 12:46:58.712 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295290 Mode  9: totalTimeGpsRunningMs                       1743091
2019-07-11 12:46:58.712 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295291 Mode  9: totalTimeNotRunningGpsMs                     -15166
2019-07-11 12:46:58.713 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295292 Mode  9: totalSuccessfulGpsTries                          61
2019-07-11 12:46:58.714 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295293 Mode  9: totalGpsTries                                 11347
2019-07-11 12:46:58.715 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295294 Mode  9: longTimeWanted                                73639
2019-07-11 12:46:58.716 19624-20311/com.rareventure.gps2 W/GpsTrailerTestUtil:   295295 Mode  9: shortTimeWanted                               30000
```

**Comment by Kright** [comment]
I have made a pull request: https://github.com/redfish64/TinyTravelTracker/pull/94

**Comment by redfish64** [comment]
Nice fix! I put it into the next version, 1.1.46
